lib/ecto/type.ex

```diff
@@ -933,6 +877,15 @@ defmodule Ecto.Type do
       :error -> {:ok, value}
     end
   end
+  def adapter_dump(adapter, {:parameterized, type, opts}, value) do
+    base_type = type.type(opts)
+    with {:ok, parameterized_value} <- type.dump(value, opts),
+         {:ok, final_value} <- adapter_dump(adapter, base_type, parameterized_value)     do
+      {:ok, final_value}
+    else
+      :error -> :error
+    end
+  end
```

Comment:
We actually have an error with that. How about just remove it as it does not replaces `{:embed, _}` match?